aptos-move/block-executor/src/view.rs

```diff
@@ -1267,27 +1266,31 @@ impl<'a, T: Transaction, S: TStateView<Key = T::Key>, X: Executable> LatestView<
                     if !resources_needing_delayed_field_exchange {
                         return None;
                     }
-                    if let Ok(Some(metadata)) = self.get_resource_state_value_metadata(key) {
-                        if let Ok(GroupReadResult::Size(group_size)) =
-                            unsync_map.get_group_size(key)
-                        {
-                            return Some(Ok((key.clone(), (metadata, group_size.get()))));
-                        } else {
-                            // TODO[agg_v2](cleanup): `get_group_size` can fail on group tag serialization. Do
-                            //       we want to propagate this error? This is somewhat an invariant
-                            //       violation so PanicError is also ok?
```

Comment:
@gelash I am propagating the VALUE_SERIALIZATION_ERROR now, that seems like a right thing to do, correct?